Detectors/GlobalTracking/include/GlobalTracking/TrackCuts.h

```diff
@@ -0,0 +1,128 @@
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file TrackCuts.h
+/// \brief Class to perform track cuts
+/// \author amelia.lindner@cern.ch
+
+#ifndef ALICEO2_TRACKCUTS_H
+#define ALICEO2_TRACKCUTS_H
+
+#include "Framework/Logger.h"
+#include "Framework/DataTypes.h"
+#include "ReconstructionDataFormats/Track.h"
+#include "DataFormatsGlobalTracking/RecoContainer.h"
+#include "ReconstructionDataFormats/MatchInfoTOF.h"
+#include "DataFormatsTPC/TrackTPC.h"
+#include "DataFormatsTRD/TrackTRD.h"
+#include "GlobalTracking/TrackMethods.h"
+#include "DetectorsBase/Propagator.h"
+#include "ReconstructionDataFormats/VtxTrackIndex.h"
+#include <set>
+#include <vector>
+#include "Rtypes.h"
+#include <gsl/span>
+namespace o2
+{
+using GID = o2::dataformats::GlobalTrackID;
+using GIndex = o2::dataformats::VtxTrackIndex;
+class TrackCuts
+{
+ public:
+  //////////////////////////////// O2 ////////////////////////////////
+  bool isSelected(GID trackIndex, o2::globaltracking::RecoContainer& data)
+  {
+    o2::track::TrackParCov trk;
+    auto contributorsGID = data.getSingleDetectorRefs(trackIndex);
+    auto src = trackIndex.getSource(); // make selections depending on track source
+    // ITS tracks
+    if (contributorsGID[GIndex::Source::ITS].isIndexSet()) { // ITS tracks selection
+      isBarrelTrack = true;
+      const auto& itsTrk = data.getITSTrack(contributorsGID[GID::ITS]);
+      int ITSnClusters = itsTrk.getNClusters();
+      float ITSchi2 = itsTrk.getChi2();
+      float itsChi2NCl = ITSnClusters != 0 ? ITSchi2 / (float)ITSnClusters : 0;
+      uint8_t itsClusterMap = itsTrk.getPattern();
+      if (ITSnClusters <= mMinNClustersITS ||
+          itsChi2NCl >= mMaxChi2PerClusterITS ||
+          TrackMethods::FulfillsITSHitRequirements(itsClusterMap, mRequiredITSHits) == false) {
+        return false;
+      }
+    }
+    // TPC tracks
+    if (contributorsGID[GIndex::Source::TPC].isIndexSet()) {
+      isBarrelTrack = true;
+      const auto& tpcTrk = data.getTPCTrack(contributorsGID[GID::TPC]);
+      // uint8_t tpcNClsShared, tpcNClsFound, tpcNClsCrossed, tpcNClsFindable, tpcChi2NCl;
+      // tpcNClsFindable = tpcTrk.getNClusters();
+      // tpcChi2NCl = tpcTrk.getNClusters() ? tpcTrk.getChi2() / tpcTrk.getNClusters() : 0;
+      // o2::TrackMethods::countTPCClusters(tpcTrk, data.getTPCTracksClusterRefs(), data.clusterShMapTPC, data.getTPCClusters(), tpcNClsShared, tpcNClsFound, tpcNClsCrossed);
+      // double tpcCrossedRowsOverFindableCls = tpcNClsCrossed / tpcNClsFindable;
+      math_utils::Point3D<float> v{};
+      std::array<float, 2> dca;
+      if (tpcTrk.getPt() < mPtTPCCut ||
+          std::abs(tpcTrk.getEta()) > mEtaTPCCut ||
+          tpcTrk.getNClusters() < mNTPCClustersCut ||
```

Comment:
I don't see where mNTPCClustersCut is assigned now?

Reply:
@shahor02  it is initialised here: https://github.com/alindner14/AliceO2/blob/dev/Detectors/GlobalTracking/include/GlobalTracking/TrackCuts.h#L88